pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -285,10 +285,14 @@ func (c *Controller) GetService(hostname host.Name) *model.Service {
 	return out
 }
 
+// mergeService only merges two clusters' k8s services
 func mergeService(dst, src *model.Service, srcRegistry serviceregistry.Instance) {
+	if !src.Ports.Equals(dst.Ports) {
+		log.Debugf("service %s defined from cluster %s is different from others", src.Hostname, srcRegistry.Cluster())
+	}
 	// Prefer the k8s HostVIPs where possible
 	clusterID := srcRegistry.Cluster()
-	if srcRegistry.Provider() == provider.Kubernetes || len(dst.ClusterVIPs.GetAddressesFor(clusterID)) == 0 {
+	if len(dst.ClusterVIPs.GetAddressesFor(clusterID)) == 0 {
```

Comment:
why this change?

Reply:
This is a redudant check, the function merge services only from k8s registry.
I also wanted to remove `len(dst.ClusterVIPs.GetAddressesFor(clusterID)) == 0` check